pkg/eventemitter/eventemitter.go

```diff
@@ -0,0 +1,91 @@
+package eventemitter
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/names"
+)
+
+type EventEmitter interface {
+	Init(mgr manager.Manager)
+	EmitEvent(object runtime.Object, eventType, reason, msg string)
+	EmitModifiedForConfig(configName string)
+	EmitProgressingForConfig(configName string)
+	EmitFailingForConfig(configName, reason, message string)
+	EmitAvailableForConfig(configName string)
+}
+
+type eventEmitter struct {
+	recorder record.EventRecorder
+	client   client.Client
+}
+
+func New(mgr manager.Manager) EventEmitter {
+	var evntEmtr EventEmitter = &eventEmitter{
+		client: mgr.GetClient(),
+	}
+
+	evntEmtr.Init(mgr)
+	return evntEmtr
+}
+
+func (ee *eventEmitter) Init(mgr manager.Manager) {
+	ee.recorder = mgr.GetEventRecorderFor(names.CLUSTER_NETWORK_ADDONS_OPERATOR_NAME)
+}
+
+func (ee eventEmitter) EmitEvent(object runtime.Object, eventType, reason, msg string) {
+	if object != nil {
+		ee.recorder.Event(object, eventType, reason, msg)
+	}
+}
+
+func (ee eventEmitter) EmitProgressingForConfig(configName string) {
+	config := ee.getConfigForEmitter(configName)
+	if config != nil {
```

Comment:
The `config != nil` is already check at `EmitEvent`  we can remove this from this function and it's friends.

Reply:
Tnx. 
Done.